### PR TITLE
[T2A2][T16-A1]Lai Zhiyuan

### PR DESCRIPTION
**Pull request by zyjames**
Ready for review

### CODE REVIEW

File: src/seedu/addressbook/AddressBook.java [hunk 1]
```diff
@@ -176,15 +162,11 @@
      */
     private static ArrayList<HashMap<PersonProperty, String>> latestPersonListingView = getAllPersonsInAddressBook(); // initial view is of all
 
-    /**
-     * The path to the file used for storing person data.
-     */
+    /** The path to the file used for storing person data. */
     private static String storageFilePath;
     
-    /**
```
Comment (okkhoy) [inline src/seedu/addressbook/AddressBook.java]: encourage you to try some more meaningful code improvements than just reformatting the comments.

File: src/seedu/addressbook/AddressBook.java [hunk 2]
```diff
@@ -250,17 +229,13 @@ private static void echoUserCommand(String userCommand) {
...
-            showToUser(MESSAGE_INVALID_PROGRAM_ARGS);
-            exitProgram();
-        }
-
-        if (args.length == 1) {
-            setupGivenFileForStorage(args[0]);
-        }
-
-        if(args.length == 0) {
+    	if (args.length == 0) {
```
Comment (okkhoy) [inline src/seedu/addressbook/AddressBook.java:232]: can think of removing magic numbers here.

File: src/seedu/addressbook/AddressBook.java [hunk 3]
```diff
@@ -250,17 +229,13 @@ private static void echoUserCommand(String userCommand) {
...
             setupDefaultFileForStorage();
+        } else if (args.length == 1) {
+        	setupGivenFileForStorage(args[0]);
```
Comment (okkhoy) [inline src/seedu/addressbook/AddressBook.java:235]: indentation is off here. we require you to use 4 spaces for indentation.
Tip: you can get eclipse to do this for you, check out how!

File: src/seedu/addressbook/AddressBook.java [hunk 4]
```diff
@@ -609,38 +581,25 @@ private static String getUserInput() {
     * ====================================================================
     */
 
-    /**
-     * Shows a message to the user
-     */
-    private static void showToUser(String[] message) {
+    /** Shows a message to the user */
+    private static void showToUser(String... message) {
```
Comment (okkhoy) [inline src/seedu/addressbook/AddressBook.java:585]: good use of varargs!

File: src/seedu/addressbook/AddressBook.java [hunk 5]
```diff
@@ -1147,14 +1104,14 @@ private static String getUsageInfoForExitCommand() {
      */
 
     /**
-     * Removes sign(p/, d/, etc) from parameter string
+     * Removes prefix(p/, d/, etc) from parameter string
```
Comment (okkhoy) [inline src/seedu/addressbook/AddressBook.java:1107]: notice that the comment still uses `p/, d/` etc., related to address book.
the exercise was to make this method as generic as possible. the same has to be reflected in the comments as well

File: src/seedu/addressbook/AddressBook.java [hunk 6]
```diff
@@ -1147,14 +1104,14 @@ private static String getUsageInfoForExitCommand() {
...
-     * @param sign  Parameter sign to be removed
-     * @return  string without the sign
+     * @param fullString  Parameter as a string
+     * @param prefix  Parameter sign to be removed
+     * @return  String without the prefix
      */
-    private static String removePrefixSign(String s, String sign) {
-        return s.replace(sign, "");
+    private static String removePrefix(String fullString, String prefix) {
+        return fullString.replace(prefix, "");
```
Comment (okkhoy) [inline src/seedu/addressbook/AddressBook.java:1114]: hmm, this is still the same as the existing solution. 
please take a careful look at the activity description for this exercise.

**Comment by okkhoy** [comment]
@zyjames some comments added. please take a look, ack and close the PR.

Avoid accumulating unrelated changes into a single PR. 
Tip: If you are currently on `branch1` and want to create a new `branch2`, you need to switch back to `master` branch before creating `branch2`. If not, changes in `branch1` will appear in the PR created from `branch2` later.


on an unrelated note, please upload your photo to github profile to help us identify :-)

File: src/seedu/addressbook/AddressBook.java (hunk 6 above)
Comment (zyjames) [reply src/seedu/addressbook/AddressBook.java:1114]: I have corrected the method! Thanks for pointing out the mistakes

**Pull request closed without merging**